### ISSUE TITLE
Valid Vagrantfile code is caught

### ISSUE THREAD

**Issue by purpleidea** [YorickPeterse/ruby-lint#152]
I decided to try using ruby-lint to clean up a Vagrantfile. This good news is that it found a bunch of legit issues, and I fixed those! Thanks ruby-lint. The bad news is that it's still finding some issues which are valid Vagrantfile, probably due the the fact that the Vagrant environment has the imported vagrant stuff.

If there is a way to make this work, I'd love to hear it! The Vagrantfile I'm using is here:
https://github.com/purpleidea/oh-my-vagrant/blob/master/vagrant/Vagrantfile

Thanks!

**Comment by purpleidea** [comment]
BTW valid Vagrant code caught includes these lines:

```
Vagrantfile: error: line 797, column 1: undefined constant Vagrant
Vagrantfile: error: line 939, column 10: undefined constant Vagrant::Errors::VagrantError
Vagrantfile: error: line 939, column 10: undefined constant Vagrant::Errors
Vagrantfile: error: line 939, column 10: undefined constant Vagrant
```

As well as these:
```
Vagrantfile: error: line 134, column 29: undefined constant XDG
```

**Comment by YorickPeterse** [comment]
This is due to the lack of a definitions file for Vagrant as well as due to Vagrant exposing methods to the global namespace. See the following for more info on this:

* https://github.com/YorickPeterse/ruby-lint/issues/129
* https://github.com/YorickPeterse/ruby-lint/issues/90
* https://github.com/YorickPeterse/ruby-lint/issues/151

There's sadly not much that can be done at this time to resolve this problem.

**Comment by purpleidea** [comment]
@YorickPeterse I figured as much. That's too bad. Since it's for the Vagrant case (and perhaps others) it's always a short list of problematic constants which trip the checker, is it possible to have a -exclude-constant VAGRANT sort of param, so I can manually exclude the known valid ones?

Thanks for your time!

**Comment by YorickPeterse** [comment]
Ignoring constants is currently not possible sadly, though any pull requests to implement this are more than welcome.

**Comment by purpleidea** [comment]
@YorickPeterse fair enough!

In the meantime, I've implemented this as a list of -a flags to turn everything else on:
https://github.com/purpleidea/oh-my-vagrant/blob/master/test.sh#L15

If someone improves upon this in the future, I'd love to hear about it.

Thanks!
